thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/RedisCacheDAO.java

```diff
@@ -0,0 +1,273 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.thirdeye.detection.cache;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.pinot.thirdeye.anomaly.utils.ThirdeyeMetricsUtil;
+import org.apache.pinot.thirdeye.detection.ConfigUtils;
+import org.redisson.Redisson;
+import org.redisson.api.RTimeSeries;
+import org.redisson.api.RedissonClient;
+import org.redisson.client.RedisConnectionException;
+import org.redisson.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * DAO class used to fetch data from Redis.
+ */
+
+public class RedisCacheDAO implements CacheDAO {
+
+	private static final String HOSTS = "hosts";
+	private static final String AUTH_USERNAME = "authUsername";
+	private static final String AUTH_PASSWORD = "authPassword";
+	public static final String IDLE_CONNECTION_TIMEOUT = "idleConnectionTimeout";
+	public static final String CONNECT_TIMEOUT = "connectTimeout";
+	public static final String TIMEOUT = "timeout";
+	public static final String RETRY_ATTEMPTS = "retryAttempts";
+	public static final String RETRY_INTERVAL = "retryInterval";
+	public static final String PING_CONNECTION_INTERVAL = "pingConnectionInterval";
+	public static final String KEEP_ALIVE = "keepAlive";
+	public static final String TCP_NO_DELAY = "tcpNoDelay";
+
+	public static final String DEFAULT_HOST = "redis://127.0.0.1:6379";
+	public static final String DEFAULT_AUTH_USERNAME = "";
+	public static final String DEFAULT_AUTH_PASSWORD = "";
+	public static final int DEFAULT_IDLE_CONNECTION_TIMEOUT = 10000;
+	public static final int DEFAULT_CONNECT_TIMEOUT = 10000;
+	public static final int DEFAULT_TIMEOUT = 3000;
+	public static final int DEFAULT_RETRY_ATTEMPTS = 3;
+	public static final int DEFAULT_RETRY_INTERVAL = 1500;
+	public static final int DEFAULT_PING_CONNECTION_INTERVAL = 30000;
+	public static final boolean DEFAULT_KEEP_ALIVE = false;
+	public static final boolean DEFAULT_TCP_NO_DELAY = false;
+
+	// private static final String SERIALIZATION = "serialization";
+	// public static final String DEFAULT_SERIALIZATION = "json";
+	// public static final String DEFAULT_ALLOWED_TYPE = "aln,var";
+
+	private static final Logger LOG = LoggerFactory.getLogger(RedisCacheDAO.class);
+
+	private Config redissonConfig;
+	private RedissonClient redissonClient;
+	private boolean redisState;
+
+	// Eg: KEY = metricId:2:dimensionKey:3158902058
+	private static final String KEY_PREFIX = "metricId:";
```

Comment:
can you change this to `thirdeyeMetricId`?

Reply:
done